src/gsy_e/models/state.py

```diff
@@ -894,19 +896,109 @@ def get_results_dict(self, current_time_slot: DateTime) -> dict:
+        return self._min_energy_demand_kWh.get(time_slot, 0)
+
+    def get_max_energy_demand_kWh(self, time_slot: DateTime) -> float:
+        """Return the maximal energy demanded for a given time slot."""
+        return self._max_energy_demand_kWh.get(time_slot, 0)
+
+    def get_temp_decrease_K(self, time_slot: DateTime) -> float:
+        """Return the temperature decrease for a given time slot."""
+        return self._temp_decrease_K.get(time_slot, 0)
+
+    def get_temp_increase_K(self, time_slot: DateTime) -> float:
+        """Return the temperature increase for a given time slot."""
+        return self._temp_increase_K.get(time_slot, 0)
+
     def get_state(self) -> Dict:
-        """To be implemented in the frame of GSYE-426"""
-        return {}
+        return {
+            "storage_temp_C": convert_pendulum_to_str_in_dict(self._storage_temp_C),
+            "temp_decrease_K": convert_pendulum_to_str_in_dict(self._temp_decrease_K),
+            "temp_increase_K": convert_pendulum_to_str_in_dict(self._temp_increase_K),
+            "min_energy_demand_kWh": convert_pendulum_to_str_in_dict(self._min_energy_demand_kWh),
+            "max_energy_demand_kWh": convert_pendulum_to_str_in_dict(self._max_energy_demand_kWh),
+        }
 
     def restore_state(self, state_dict: Dict):
-        """To be implemented in the frame of GSYE-426"""
+        self._storage_temp_C = convert_pendulum_to_str_in_dict(state_dict["storage_temp_C"])
+        self._temp_decrease_K = convert_pendulum_to_str_in_dict(state_dict["temp_decrease_K"])
+        self._temp_increase_K = convert_pendulum_to_str_in_dict(state_dict["temp_increase_K"])
+        self._min_energy_demand_kWh = convert_pendulum_to_str_in_dict(
+            state_dict["min_energy_demand_kWh"])
+        self._max_energy_demand_kWh = convert_pendulum_to_str_in_dict(
+            state_dict["max_energy_demand_kWh"])
 
     def delete_past_state_values(self, current_time_slot: DateTime):
-        """To be implemented in the frame of GSYE-426"""
+        if not current_time_slot or constants.RETAIN_PAST_MARKET_STRATEGIES_STATE:
+            return
+        self._delete_time_slots(self._min_energy_demand_kWh,
+                                self._last_time_slot(current_time_slot))
+        self._delete_time_slots(self._max_energy_demand_kWh,
+                                self._last_time_slot(current_time_slot))
+        self._delete_time_slots(self._storage_temp_C,
+                                self._last_time_slot(current_time_slot))
+        self._delete_time_slots(self._temp_increase_K,
+                                self._last_time_slot(current_time_slot))
+        self._delete_time_slots(self._temp_decrease_K,
+                                self._last_time_slot(current_time_slot))
+
+    def get_results_dict(self, current_time_slot: DateTime) -> Dict:
+        return {
+            "storage_temp_CC": self.get_storage_temp_C(current_time_slot),
```

Comment:
Could you please describe what `CC` and `KK` are? Thanks.

Reply:
Ah, this is remnant of testing, let me fix this.

Reply:
[574c0ce](https://github.com/gridsingularity/gsy-e/pull/1581/commits/574c0ce6174241e6c983a8740d9162618124d2e7)